### PR TITLE
feat(): adding ci pipeline

### PR DESCRIPTION
**Pull request by artbegolli**
CI pipeline and linting fixes

### CODE REVIEW

**Comment by VaibhavPage** [comment]
For WIP, there is a dropdown you can select on `Create Merge Request` button.

**Comment by artbegolli** [comment]
This should be good to go - push steps are triggered when a tag is dropped on the repo

File: pkg/docker/docker_test.go [hunk 1]
```diff
@@ -58,8 +58,11 @@ func TestDocker_Pull(t *testing.T) {
 	assert.Equal(t, nil, err)
 }
 
-func (t testClient) ImageBuild(ctx context.Context, context io.Reader, options types.ImageBuildOptions) (io.ReadCloser, error) {
-	return nil, nil
+func (t testClient) ImageBuild(ctx context.Context, context io.Reader, options types.ImageBuildOptions) (types.ImageBuildResponse, error) {
```
Comment (VaibhavPage) [inline pkg/docker/docker_test.go]: Can we not have the ImageBuildResponse in the types.go? Can;t this be ImageBuildResponse from `docker`?

File: common/util.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ import (
 
 // GetClientConfig return rest config, if path not specified, assume in cluster config
 func GetClientConfig(kubeconfig string) (*rest.Config, error) {
-	if kubeconfig != "" {
+	//if kubeconfig != "" {
```
Comment (VaibhavPage) [inline common/util.go]: why is this commented?

File: pkg/docker/docker_test.go (hunk 1 above)
Comment (artbegolli) [reply pkg/docker/docker_test.go]: This imagebuildresponse is from docker, we can probably get rid of our custom defined one.

File: common/util.go (hunk 2 above)
Comment (artbegolli) [reply common/util.go]: This came in as part of the operator logic and resulted in tests failing.

Comment (VaibhavPage) [reply common/util.go]: which test failed?

Comment (artbegolli) [reply common/util.go]: If I remember right it was a dependency issue with clientcmd, not at my machine so can't take a look

Comment (artbegolli) [reply common/util.go]: Will have a look in the morning

Comment (VaibhavPage) [reply common/util.go]: Cool. The output of this function depends on the environment you are trying to run the tests in. So I am inclined towards not testing this one function.

Comment (artbegolli) [reply common/util.go]: Sounds good, Ill add an exclusion so we can uncomment it

Comment (artbegolli) [reply common/util.go]: This has been resolved - I've updated Gopkg.lock to reflect the clientcmd dependency

File: pkg/docker/docker_test.go (hunk 1 above)
Comment (artbegolli) [reply pkg/docker/docker_test.go]: We are no longer using our own ImageBuildResponse. All commands to docker/buildah return ReadClosers which can then be read by the user. 

The ImageBuildResponse here is to implement the docker API client for mocking purposes.

**Pull request merged**